src/npcmove.cpp

```diff
@@ -3716,17 +3716,18 @@ void npc::mug_player( player &mark )
         value_mod -= ( ( 8 - op_of_u.value ) * .07 );
     }
     double best_value = minimum_item_value() * value_mod;
-    int item_index = INT_MIN;
+    item *to_steal = nullptr;
     invslice slice = mark.inv.slice();
-    for( size_t i = 0; i < slice.size(); i++ ) {
-        if( value( slice[i]->front() ) >= best_value &&
-            can_pickVolume( slice[i]->front(), true ) &&
-            can_pickWeight( slice[i]->front(), true ) ) {
-            best_value = value( slice[i]->front() );
-            item_index = i;
+    for( std::list<item> *stack : slice ) {
+        item *front_stack = &stack->front();
```

Comment:
Why a pointer? This value is used as pointer at only one place, all other usage uses it as reference anyway. So making this a reference will remove several unnecessary `*` from the code.

Reply:
because references cannot be null. I suppose i can add in a reference on the inside of the for loop to decrease the amount of calls to operator* if that's what you're after?

Reply:
`front_stack` is never set to null. It is always set to a valid non-null pointer.

Reply:
ah my bad, i was thinking you were talking about `to_steal` for some reason. i'll get to changing that soon.